src/wrapper.js

```diff
@@ -0,0 +1,162 @@
+wrapper.runSnippet = function runSnippet (walkmeSnippetUrl) {
+  let walkme = document.createElement('script');
+  let script;
+  walkme.type = 'text/javascript';
+  walkme.async = true;
+  walkme.src = walkmeSnippetUrl;
+  script = document.getElementsByTagName('script')[0];
+  script.parentNode.insertBefore(walkme, script);
+  window._walkmeConfig = { smartLoad: true };
+};
+
+wrapper.startWalkthrough = afterApiLoad(function (id) {
+  WalkMeAPI.startWalkthruById(id);
+});
+
+wrapper.isLoaded = afterApiLoad(() => true);
+
+let eventSubscribers = Object.keys(wrapper.Event).reduce((subscribers, event) => {
+  subscribers[event] = [];
+  return subscribers;
+}, {});
+
+/**
+ * Listen to a WalkMe event
+ *
+ * @param  {String} eventName - the WalkMe event
+ * @param  {Function} handler - the method to execute when this event is received
+ */
+wrapper.on = afterApiLoad(function (eventName, handler) {
+  window.menu_event = window.menu_event || function onMenuEvent (eventData) {
+    eventSubscribers[eventData.Type].forEach(subscriber => {
+      subscriber(eventData);
+    });
+  };
+
+  window.walkme_event = window.walkme_event || function onWalkMeEvent (eventData) {
+    eventSubscribers[eventData.Type].forEach(subscriber => {
+      subscriber(eventData);
+    });
+  };
+
+  eventSubscribers[eventName].push(handler);
+});
+
+const lastStepMemo = {};
+function isLastStep (walkthruId, stepIndex) {
+  if (walkthruId in lastStepMemo) {
+    return lastStepMemo[walkthruId] === stepIndex;
+  }
+
+  let isUndocumentedApiAvailable = _walkMe && typeof _walkMe.getTutorials === 'function';
+
+  if (isUndocumentedApiAvailable) {
+    let tutorial = find(_walkMe.getTutorials(), {Id: walkthruId});
+    let sortedSteps = sortBy(tutorial.Steps, 'StepIndex');
+    let lastStep = tutorial && findLast(sortedSteps, step => {
+      return !step.IsSkippable && step.Status === 1 && step.Type !== StepType.SuperStep;
+    });
+
+    if (typeof lastStep !== 'undefined') {
+      lastStepMemo[walkthruId] = lastStep.StepIndex;
+      return lastStep.StepIndex === stepIndex;
+    }
+  }
+
+  return false;
+}
+
+wrapper.onWalkthroughCompleted = function onWalkthroughCompleted (handler) {
+  function onAllStepsFinished (eventData) {
+    if (isLastStep(eventData.WalkthruId, eventData.StepIndex)) {
+      handler(eventData);
+    }
+  }
+
+  wrapper.on(wrapper.Event.TaskCompleted, handler);
+  wrapper.on(wrapper.Event.NextButtonClicked, onAllStepsFinished);
+  wrapper.on(wrapper.Event.DoneButtonClicked, onAllStepsFinished);
+};
+
+/**
+ * addVariables attaches a "CustomWalkMeCompanyVariables" object
+ * to window with the given values.
+ *
+ * This method can be called multiple times and the new variables
+ * will be merged with the existing variables
+ *
+ * @param {Object} variables - a map of variable name to value
+ */
+wrapper.addVariables = function addVariables (variables) {
+  window.CustomWalkMeCompanyVariables = assign(window.CustomWalkMeCompanyVariables, variables);
```

Comment:
we stick these right on window because walkme expects them there?

Reply:
yup, they need to be global so we can reference them in the WalkMe UI when creating the walkthrough